Mathlib/CategoryTheory/ComposableArrows.lean

```diff
@@ -257,6 +264,167 @@ lemma ext₁ {F G : ComposableArrows C 1}
+      · simp at hjk
+      · dsimp
+        rw [F.map_id, comp_id]
+      · rfl
+    · obtain _ | _ | k := k
+      · simp [Fin.ext_iff] at hjk
+      · simp [Fin.le_def, Nat.succ_eq_add_one] at hjk
+      · dsimp
+        rw [assoc, ← F.map_comp, homOfLE_comp]
+  · obtain _ | j := j
+    · simp [Fin.ext_iff] at hij
+    · obtain _ | k := k
+      · simp [Fin.ext_iff] at hjk
+      · dsimp
+        rw [← F.map_comp, homOfLE_comp]
+
+end Precomp
+
+/-- "Precomposition" of `F : ComposableArrows C n` by a morphism `f : X ⟶ F.left`. -/
+@[simps]
+def precomp {X : C} (f : X ⟶ F.left) : ComposableArrows C (n + 1) where
+  obj := Precomp.obj F X
+  map g := Precomp.map F f _ _ (leOfHom g)
+  map_id := Precomp.map_id F f
+  map_comp g g' := (Precomp.map_comp F f (leOfHom g) (leOfHom g'))
+
+/-- Constructor for `ComposableArrows C 2`. -/
+@[simp]
+def mk₂ {X₀ X₁ X₂ : C} (f : X₀ ⟶ X₁) (g : X₁ ⟶ X₂) : ComposableArrows C 2 :=
+  (mk₁ g).precomp f
+
+/-- Constructor for `ComposableArrows C 3`. -/
+@[simp]
+def mk₃ {X₀ X₁ X₂ X₃ : C} (f : X₀ ⟶ X₁) (g : X₁ ⟶ X₂) (h : X₂ ⟶ X₃) : ComposableArrows C 3 :=
+  (mk₂ g h).precomp f
+
+section
+
+variable {X₀ X₁ X₂ X₃ X₄ : C} (f : X₀ ⟶ X₁) (g : X₁ ⟶ X₂) (h : X₂ ⟶ X₃) (i : X₃ ⟶ X₄)
+
+/-! These examples are meant to test the good definitional properties of `precomp`,
+and that `dsimp` can see through. -/
+
+example : map' (mk₂ f g) 0 1 = f := by dsimp
+example : map' (mk₂ f g) 1 2 = g := by dsimp
+example : map' (mk₂ f g) 0 2 = f ≫ g := by dsimp
+example : (mk₂ f g).hom = f ≫ g := by dsimp
+example : map' (mk₂ f g) 0 0 = 𝟙 _ := by dsimp
+example : map' (mk₂ f g) 1 1 = 𝟙 _ := by dsimp
+example : map' (mk₂ f g) 2 2 = 𝟙 _ := by dsimp
+
+example : map' (mk₃ f g h) 0 1 = f := by dsimp
+example : map' (mk₃ f g h) 1 2 = g := by dsimp
+example : map' (mk₃ f g h) 2 3 = h := by dsimp
+example : map' (mk₃ f g h) 0 3 = f ≫ g ≫ h := by dsimp
+example : (mk₃ f g h).hom = f ≫ g ≫ h := by dsimp
+example : map' (mk₃ f g h) 0 2 = f ≫ g := by dsimp
+example : map' (mk₃ f g h) 1 3 = g ≫ h := by dsimp
+
+end
+
+/-- The functor `Fin n ⥤ Fin (n + 1)` which sends `i` to `i.succ`. -/
+@[simps]
+def _root_.Fin.succFunctor (n : ℕ) : Fin n ⥤ Fin (n + 1) where
+  obj i := i.succ
+  map {i j} hij := homOfLE (Fin.succ_le_succ_iff.2 (leOfHom hij))
+
+/-- The functor `ComposableArrows C (n + 1) ⥤ ComposableArrows C n` which forgets
+the first arrow. -/
+@[simps!]
+def δ₀Functor : ComposableArrows C (n + 1) ⥤ ComposableArrows C n :=
+  (whiskeringLeft _ _ _).obj (Fin.succFunctor _)
+
+/-- The `ComposableArrows C n` obtained by forgetting the first arrow. -/
+abbrev δ₀ (F : ComposableArrows C (n + 1)) := δ₀Functor.obj F
+
+@[simp]
+lemma precomp_δ₀ {X : C} (f : X ⟶ F.left) : (F.precomp f).δ₀ = F := rfl
+
 end ComposableArrows
 
+namespace Nerve
+
+variable {C}
+
+lemma δ₀_eq_composableArrows_δ₀ {n : ℕ}
+    (x : (nerve C).obj (Opposite.op (SimplexCategory.mk (n + 1)))) :
+    (nerve C).δ (0 : Fin (n + 2)) x = ComposableArrows.δ₀ x := rfl
+
+end Nerve
```

Comment:
I wonder whether we should revert the import. If so, then we can do that in another PR.

Reply:
@jcommelin  Do you mean have `Nerve` import `ComposableArrows` rather than the other direction, so that we may use `ComposableArrows` without relying on the whole simplicial stuff?

Reply:
Exactly! And it would also allow us to use `ComposableArrows` from the start in setting up API for `Nerve`.

Reply:
As bors failed, I shall do it in this PR.

Reply:
This is done in https://github.com/leanprover-community/mathlib4/pull/8146/commits/6a9f2359fec047680e7558b306bbed6bec9ce083